gpkit/interactive/svg.py

```diff
@@ -0,0 +1,78 @@
+import svgwrite
+from svgwrite import cm
+"""
+Contains all svgwrite dependent methods
+"""
+
+def make_diagram(sol, depth, filename, sidelength, height, input_dict):
+    """
+    method called to make the diagram - calls importsvgwrite from interactive
+    to import svgwrite, calls all necessary follow on methods and sets
+    important variables
+    """
+    #extract the total breakdown value for scaling purposes
+    total = sol(input_dict.keys()[0])
+    #depth of each breakdown level
+    elementlength = (sidelength/depth)
+    dwg = svgwrite.Drawing(filename, debug=True)
+    dwgrecurse(input_dict, (2, 2), -1, sol, elementlength, height, total, depth, dwg)
+    #save the drawing at the conlusion of the recursive call
+    dwg.save()
+
+    return dwg
+
+def dwgrecurse(input_dict, initcoord, currentlevel, sol, elementlength, sheight, total, depth, dwg):
+    """
+    recursive function to divide widnow into seperate units to be drawn and
+    calls the draw function
+    """
+    totalheight = 0
+    currentlevel = currentlevel+1
+    for key in sorted(input_dict):
+        height = int(round((((sheight/total)*sol(key)))))
+        currentcoord = (initcoord[0], initcoord[1]+totalheight)
+        drawsegment(key, height, currentcoord, dwg, elementlength)
+        totalheight = totalheight+height
+        if isinstance(input_dict[key], dict):
```

Comment:
@bqpd oh wow sorry I'm an idiot and totally misunderstood what the first comment was about. For some reason my git is saying you'r commenting on an outdated diff...but best as I can tell this is the most current thing there is, it seems the same as what I have locally